src/main/scala/beam/router/R5RoutingApp.scala

```diff
@@ -0,0 +1,184 @@
+import io.circe.generic.decoding.DerivedDecoder
+import io.circe.generic.encoding.DerivedObjectEncoder
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+import io.circe.syntax._
+import io.circe.{Decoder, Encoder, HCursor, Json}
+import org.matsim.api.core.v01.Id
+import org.matsim.core.config.groups.TravelTimeCalculatorConfigGroup
+import org.matsim.vehicles.Vehicle
+import shapeless.Lazy
+
+import scala.concurrent.Await
+import scala.concurrent.duration._
+import scala.reflect.ClassTag
+
+class IdFormat[T](implicit val ct: ClassTag[T]) extends Encoder[Id[T]] with Decoder[Id[T]] {
+  override def apply(o: Id[T]): Json = Json.fromString(o.toString)
+
+  override def apply(c: HCursor): Result[Id[T]] = {
+    c.as[String].map { id =>
+      val clazz = ct.runtimeClass.asInstanceOf[Class[T]]
+      Id.create(id, clazz)
+    }
+  }
+}
+
+object LocationFormat extends Encoder[Location] with Decoder[Location] {
+  override def apply(a: Location): Json = Json.obj(
+    ("x", Json.fromString(a.getX.toString)),
+    ("y", Json.fromString(a.getY.toString))
+  )
+
+  override def apply(c: HCursor): Result[Location] = {
+    for {
+      x <- c.downField("x").as[String]
+      y <- c.downField("y").as[String]
+    } yield {
+      new Location(x.toDouble, y.toDouble)
+    }
+  }
+}
+
+class Format[T](implicit decode: Lazy[DerivedDecoder[T]], encode: Lazy[DerivedObjectEncoder[T]])
+    extends Encoder[T]
+    with Decoder[T] {
+  implicit val decoder: Decoder[T] = deriveDecoder[T]
+  implicit val encoder: Encoder[T] = deriveEncoder[T]
+
+  override def apply(a: T): Json = encoder.apply(a)
+
+  override def apply(c: HCursor): Result[T] = decoder.apply(c)
+}
+
+object AllNeededFormats {
+  implicit val locationFormat = LocationFormat
+  implicit val beamModeFormat = new Format[BeamMode]
+  implicit val vehicleIdFormat = new IdFormat[Vehicle]
+  implicit val beamVehicleTypeFormat = new IdFormat[BeamVehicleType]
+
+  implicit val spaceTimeFormat = new Format[SpaceTime]
+  implicit val streetVehicleFormat = new Format[StreetVehicle]
+  implicit val householdAttributesFormat = new Format[HouseholdAttributes]
+  implicit val attributesOfIndividualFormat = new Format[AttributesOfIndividual]
+  implicit val intermodalUseFormat = new Format[IntermodalUse]
+  implicit val routingRequestFormat = new Format[RoutingRequest]
+
+  implicit val transitStopsInfoFormat = new Format[TransitStopsInfo]
+  implicit val beamPathFormat = new Format[BeamPath]
+  implicit val beamLegFormat = new Format[BeamLeg]
+  implicit val embodiedBeamLegFormat = new Format[EmbodiedBeamLeg]
+  implicit val embodiedBeamTripFormat = new Format[EmbodiedBeamTrip]
+  implicit val routingResponseFormat = new Format[RoutingResponse]
+}
+
+class RoutingHandler(val workerRouter: ActorRef) extends FailFastCirceSupport {
+  implicit val timeout: Timeout = new Timeout(10, TimeUnit.SECONDS)
+  import AllNeededFormats._
+
+  val route: Route = {
+    path("find-route") {
+      post {
```

Comment:
This is variable depending on current data, correct? Otherwise it should be a get imo

Reply:
It's a `POST` request because it has to have a body(routing request). Here is an example of curl request:
```
curl -X POST \
  http://localhost:9000/find-route \
  -H 'Content-Type: application/json' \
  -H 'Postman-Token: 793e2600-901d-4947-8a37-57d04096e465' \
  -H 'cache-control: no-cache' \
  -d '{
  "originUTM" : {
    "x" : "551626.6559302211",
    "y" : "4182124.0693679387"
  },
  "destinationUTM" : {
    "x" : "543850.7021433391",
    "y" : "4187178.598555087"
  },
  "departureTime" : 24636,
  "transitModes" : [
    {
      "TRANSIT" : {
        
      }
    }
  ],
  "streetVehicles" : [
    {
      "id" : "body-4436198",
      "vehicleTypeId" : "BODY-TYPE-DEFAULT",
      "locationUTM" : {
        "loc" : {
          "x" : "551626.6559302211",
          "y" : "4182124.0693679387"
        },
        "time" : 23736
      },
      "mode" : {
        "WALK" : {
          
        }
      },
      "asDriver" : true
    },
    {
      "id" : "dummyRH",
      "vehicleTypeId" : "CAR-TYPE-DEFAULT",
      "locationUTM" : {
        "loc" : {
          "x" : "551626.6559302211",
          "y" : "4182124.0693679387"
        },
        "time" : 24636
      },
      "mode" : {
        "CAR" : {
          
        }
      },
      "asDriver" : false
    }
  ],
  "attributesOfIndividual" : null,
  "streetVehiclesUseIntermodalUse" : {
    "AccessAndEgress" : {
      
    }
  },
  "requestId" : 934298
}'
```